src/components/AzureMapFeature/useFeature.ts

```diff
@@ -1,28 +1,36 @@
 import { useEffect } from 'react'
 import { DataSourceType, IAzureMapFeature, ShapeType, FeatureType } from '../../types'
 import { useCheckRef } from '../../hooks/useCheckRef'
+import atlas from 'azure-maps-control'
 
 export const useFeature = (
   { setCoords, setProperties }: IAzureMapFeature,
   dataSourceRef: DataSourceType | null,
   shapeRef: ShapeType | null,
   featureRef: FeatureType | null
 ) => {
-  // Simple feature's usecases and methods
-  useCheckRef<DataSourceType, FeatureType>(dataSourceRef, featureRef, (dref, fref) => {
-    dref.add(fref)
-    return () => {
-      dref.remove(fref)
-    }
-  })
+  if(dataSourceRef instanceof atlas.source.DataSource){
```

Comment:
We can't call react hooks conditionally it will trigger an eslint error, and I don't think that is safe.

Reply:
thanks for pointing out, just saw the rules of hooks

Reply:
@msasinowski: so `npm run lint` also fails at https://github.com/WiredSolutions/react-azure-maps/blob/207aeaa1f83c8283dee3a5614ce2fd54c7143b43/src/components/AzureMap/AzureMap.tsx#L65-L74 where functions (which are pure map ref modifiers and are not composed of any effects) run inside the event handler which should be ok but they are named as effects so the eslint fails on them. Do you want me to create another PR to rename those?

Reply:
I see your PR for changing that and already approved, Yes I add this kind of new config along with adding integration with the new rollup version, so errors and warnings come up after this change. Good point and that's for that.

Reply:
@msasinowski: got it, also I remove `"pre-commit": "lint-staged"` from package.json husky hooks, as it changes the auto changes the code style in every file changed and produces a lot of edits, the style is uses is a bit different to the one assumed here, like semicolons after every line etc so I guess we might also need to tweak prettier there